$NetBSD: patch-Makefile.install,v 1.8 2026/02/24 23:09:02 thor Exp $

Reform library linking commands to work for any library naming setup, ensuring
that we got the generic libopenblas.so.0 and libopenblas.so names, either as
symlink or as library file.

Remove duplicate definition of SUFFIX64.

Simplify and fix the pkg-config file for FIXED_LIBNAME.

--- Makefile.install.orig	2026-01-15 22:57:26.000000000 +0000
+++ Makefile.install
@@ -2,15 +2,11 @@ TOPDIR	= .
 export GOTOBLAS_MAKEFILE = 1
 -include $(TOPDIR)/Makefile.conf_last
 include ./Makefile.system
-LNCMD = ln -fs
 
 ifdef THELIBNAME
 LIBNAME=$(THELIBNAME)
 LIBSONAME=$(THELIBSONAME)
 endif
-ifeq ($(FIXED_LIBNAME), 1)
-LNCMD = true
-endif
 ifeq ($(INTERFACE64),1)
 USE_64BITINT=1
 endif
@@ -32,7 +28,7 @@ PKG_EXTRALIB := $(EXTRALIB)
 ifeq ($(INTERFACE64),1)
 	SUFFIX64=64
 endif
-PKGFILE="$(DESTDIR)$(OPENBLAS_PKGCONFIG_DIR)/$(LIBSONAMEBASE)$(SUFFIX64).pc"
+PKGFILE="$(DESTDIR)$(OPENBLAS_PKGCONFIG_DIR)/$(LIBNAMEBASE).pc"
 
 ifeq ($(USE_OPENMP), 1)
   ifeq ($(C_COMPILER), PGI)
@@ -124,29 +120,23 @@ ifneq ($(NO_STATIC),1)
 	@echo Copying the static library to $(DESTDIR)$(OPENBLAS_LIBRARY_DIR)
 	@install -m644 $(LIBNAME) "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
 	@cd "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)" ; \
-	$(LNCMD) $(LIBNAME) $(LIBPREFIX).$(LIBSUFFIX)
+	$(call LNCMD,$(LIBNAME),$(LIBPREFIX).$(LIBSUFFIX))
 endif
 #for install shared library
 ifneq ($(NO_SHARED),1)
 	@echo Copying the shared library to $(DESTDIR)$(OPENBLAS_LIBRARY_DIR)
-ifeq ($(OSNAME), $(filter $(OSNAME),Linux SunOS Android Haiku FreeBSD DragonFly))
+ifeq ($(OSNAME), $(filter $(OSNAME),Linux SunOS Android Haiku FreeBSD DragonFly OpenBSD NetBSD))
 	@install -m755 $(LIBSONAME) "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
 	@cd "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)" ; \
-	$(LNCMD) $(LIBSONAME) $(LIBPREFIX).so ; \
-	$(LNCMD) $(LIBSONAME) $(LIBPREFIX).so.$(MAJOR_VERSION)
-endif
-
-ifeq ($(OSNAME), $(filter $(OSNAME),OpenBSD NetBSD))
-	@cp $(LIBSONAME) "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
-	@cd "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)" ; \
-	$(LNCMD) $(LIBSONAME) $(LIBPREFIX).so
+	$(call LNCMD,$(LIBSONAME),$(LIBPREFIX).so) ; \
+	$(call LNCMD,$(LIBSONAME),$(LIBPREFIX).so.$(MAJOR_VERSION))
 endif
 ifeq ($(OSNAME), Darwin)
 	@-cp $(LIBDYNNAME) "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
 	@-install_name_tool -id "$(OPENBLAS_LIBRARY_DIR)/$(LIBPREFIX).$(MAJOR_VERSION).dylib" "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)/$(LIBDYNNAME)"
 	@cd "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)" ; \
-	$(LNCMD) $(LIBDYNNAME) $(LIBPREFIX).dylib ; \
-	$(LNCMD) $(LIBDYNNAME) $(LIBPREFIX).$(MAJOR_VERSION).dylib
+	$(call LNCMD,$(LIBDYNNAME),$(LIBPREFIX).dylib) ; \
+	$(call LNCMD,$(LIBDYNNAME),$(LIBPREFIX).$(MAJOR_VERSION).dylib)
 endif
 ifeq ($(OSNAME), WINNT)
 	@-cp $(LIBDLLNAME) "$(DESTDIR)$(OPENBLAS_BINARY_DIR)"
@@ -174,30 +164,23 @@ ifneq ($(NO_STATIC),1)
 	@echo Copying the static library to $(DESTDIR)$(OPENBLAS_LIBRARY_DIR)
 	@installbsd -c -m 644 $(LIBNAME) "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
 	@cd "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)" ; \
-	$(LNCMD) $(LIBNAME) $(LIBPREFIX).$(LIBSUFFIX)
+	$(call LNCMD,$(LIBNAME),$(LIBPREFIX).$(LIBSUFFIX))
 endif
 #for install shared library
 ifneq ($(NO_SHARED),1)
 	@echo Copying the shared library to $(DESTDIR)$(OPENBLAS_LIBRARY_DIR)
 	@installbsd -c -m 755 $(LIBSONAME) "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
 	@cd "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)" ; \
-	$(LNCMD) $(LIBSONAME) $(LIBPREFIX).so ; \
-	$(LNCMD) $(LIBSONAME) $(LIBPREFIX).so.$(MAJOR_VERSION)
+	$(call LNCMD,$(LIBSONAME),$(LIBPREFIX).so) ; \
+	$(call LNCMD,$(LIBSONAME),$(LIBPREFIX).so.$(MAJOR_VERSION))
 endif
 
 endif
 
 #Generating openblas.pc
-ifeq ($(INTERFACE64),1)
-	SUFFIX64=64
-endif
-	PKGFILE="$(DESTDIR)$(OPENBLAS_PKGCONFIG_DIR)/$(LIBSONAMEBASE)$(SUFFIX64).pc"
-
-	@echo Generating $(LIBSONAMEBASE)$(SUFFIX64).pc in "$(DESTDIR)$(OPENBLAS_PKGCONFIG_DIR)"
+	@echo Generating "$$(basename "$(PKGFILE)")" in "$$(dirname "$(PKGFILE)")"
 	@echo 'libdir='$(OPENBLAS_LIBRARY_DIR) > "$(PKGFILE)"
-	@echo 'libprefix='$(LIBNAMEPREFIX) >> "$(PKGFILE)"
-	@echo 'libnamesuffix='$(LIBNAMESUFFIX) >> "$(PKGFILE)"
-	@echo 'libsuffix='$(SYMBOLSUFFIX) >> "$(PKGFILE)"
+	@echo 'libname='$(LIBLINKPREFIX) >> "$(PKGFILE)"
 	@echo 'includedir='$(OPENBLAS_INCLUDE_DIR) >> "$(PKGFILE)"
 	@echo 'omp_opt='$(FOMP_OPT) >> "$(PKGFILE)"
 	@echo 'openblas_config= USE_64BITINT='$(INTERFACE64) 'DYNAMIC_ARCH='$(DYNAMIC_ARCH) 'DYNAMIC_OLDER='$(DYNAMIC_OLDER) 'NO_CBLAS='$(NO_CBLAS) 'NO_LAPACK='$(NO_LAPACK) 'NO_LAPACKE='$(NO_LAPACKE) 'NO_AFFINITY='$(NO_AFFINITY) 'USE_OPENMP='$(USE_OPENMP) $(TARGET) 'MAX_THREADS='$(NUM_THREADS)>> "$(PKGFILE)"
